### ISSUE TITLE
💅 Valid Astro imports fail `lint/correctness/noUnusedImports`

### ISSUE THREAD

**Issue by AdoryVo** [biomejs/biome#2233]
### Environment information

```bash
CLI:
  Version:                      1.6.3
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           windows

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         unset
  JS_RUNTIME_VERSION:           "v20.10.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "pnpm/8.15.5"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 true

Linter:
  Recommended:                  true
  All:                          false
  Rules:                        correctness/noUnusedImports = "warn"

Workspace:
  Open Documents:               0
```

### Rule name

lint/correctness/noUnusedImports

### Playground link

https://codesandbox.io/p/devbox/astro-no-unused-imports-fp8zmh?file=%2Fsrc%2F404.astro%3A5%2C4

### Expected result

The import was used validly via [Astro component syntax](https://docs.astro.build/en/basics/astro-components/#the-component-template), so a `noUnusedImports` linter warning should not be thrown.

![image](https://github.com/biomejs/biome/assets/33337855/b7b7a2d7-25a5-460c-98e8-523e6c0e3fb4)
> - Lint warning from linked playground

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
I think, for the time being, it's best to disable this rule for `.astro` files. In order to fix it, we would need to achieve full support

**Comment by bcheung** [comment]
still seems to be an issue with 2.3 with experimental full astro support when importing in frontmatter and using in html (unless i configured it wrong)
https://biomejs.dev/blog/biome-v2-3/

**Comment by dyc3** [comment]
Yes, that is to be expected for now @bcheung (hence why its experimental). As noted on [the language support page](https://biomejs.dev/internals/language-support/#html-super-languages-support), a current restriction is that we can't evaluate rules that should go across "language boundaries", and we still recommend you disable these kinds of rules for these files for the time being. It'll get better though, it's on our radar.

**Comment by CatBraaain** [comment]
FYI: As a workaround, you can disable noUnused rules only for `.astro` files.

```json
{
  "overrides": [
    {
      "includes": ["**/*.astro"],
      "linter": {
        "rules": {
          "correctness": {
            "noUnusedVariables": "off",
            "noUnusedImports": "off"
          }
        }
      }
    }
  ]
}
```

**Comment by tats-u** [comment]
> we still recommend you disable these kinds of rules for these files for the time being. It'll get better though, it's on our radar.

This should be turned them off _by default_ for the time being. The current behavior is very harmful. Who on earth wants to keep them on even when they find Biome vomits such false-positive "errors"? The frontmatter TS code in an `.astro` file is just an optional extra.
